src/steamship/base/client.py

```diff
@@ -63,10 +64,19 @@ def __init__(
         )
         self._session = Session()
         super().__init__(config=config)
-        self.switch_space(space_id=space_id, space_handle=space_handle, create_space=create_space)
+        self.switch_space(
+            space_id=space_id,
+            space_handle=space_handle,
+            create_space=create_space,
+            fetch_or_create_space=fetch_or_create_space,
+        )
 
     def switch_space(
-        self, space_id: str = None, space_handle: str = None, create_space: bool = False
+        self,
+        space_id: str = None,
```

Comment:
Do we need both space_id and space_handle? Aren't both unique to a given user? 

You can argue that space_ids are unique across users too, but from the scope of a user, that doesn't increase the utility of using it. 

Possible upside: The user cannot get a conflict by providing a `space_id` that doesn't exist and cannot be created (as the engine doesn't allow you specify the id). This is an exception flow that doesn't exist when the `space_handle` is used.

Reply:
Right now we seem to have a mix of allowing handles / ids to be used to fetch things.  I think some more consistency on this across the board would be nice, but for the switch_space use case specifically, I would be happy to drop ID from the interface.

Question though - when these are being initialized from the proxy in a plugin / app call, is the space handle passed or just the id?

Reply:
Agree that `*_id` is superfluous given that handles are the preferred way of referencing things.

IIRC both the engine and proxy support `space_handle` in addition to `space_id`, though the current client code may be passing id behind the scenes to avoid an extra db round-trip (since `id` is the fk in the db)